airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -94,14 +94,17 @@ public class ConfigRepository {
 
   private final ConfigPersistence persistence;
   private final ExceptionWrappingDatabase database;
+  private final ActorDefinitionMigrator actorDefinitionMigrator;
 
   public ConfigRepository(final ConfigPersistence persistence, final Database database) {
-    this.persistence = persistence;
-    this.database = new ExceptionWrappingDatabase(database);
+    this(persistence, database, new ActorDefinitionMigrator(new ExceptionWrappingDatabase(database)));
   }
 
-  public ConfigPersistence getConfigPersistence() {
-    return persistence;
+  @VisibleForTesting
+  ConfigRepository(final ConfigPersistence persistence, final Database database, final ActorDefinitionMigrator actorDefinitionMigrator) {
```

Comment:
Can we keep only this constructor? Otherwise it will make us to move away from Micronaut, We should update the bean of the different BeanFactories.